### PR TITLE
GHGRMHD add FD directory and basic structure

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Add GHGRMHD FD subdir
- Add reconstructor base class for GHGRMHD
- Add registration files
- Add missing link library to GHGRMHD

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/Reconstructor.hpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  /// \cond
+  explicit Reconstructor(CkMigrateMessage* msg);
+  WRAPPED_PUPable_abstract(Reconstructor);  // NOLINT
+  /// \endcond
+
+  using creatable_classes = tmpl::list<>;
+
+  virtual std::unique_ptr<Reconstructor> get_clone() const = 0;
+
+  virtual size_t ghost_zone_size() const = 0;
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/Reconstructor.hpp:40]: include for size_t

File: src/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/Reconstructor.hpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <pup.h>
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/Reconstructor.hpp]: could change this to a forward declaration

**Review by kidder — CHANGES_REQUESTED** [review]
can squash immediately

**Review by yoonso0-0 — DISMISSED** [review]
No further comments from me

**Comment by nilsdeppe** [comment]
rebased and squashed in the changes. Thanks for the reviews :)

File: src/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/Reconstructor.hpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <size_t>
```
Comment (yoonso0-0) [inline src/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/Reconstructor.hpp]: `<cstddef>`?

Comment (nilsdeppe) [reply src/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/Reconstructor.hpp]: hah... yea... :( Thanks!

**Review by kidder — APPROVED** [review]

**Pull request merged**